packages/api/app/Controllers/Http/TechnologyController.js

```diff
@@ -255,6 +270,63 @@ class TechnologyController {
 		return technology;
 	}
 
+	/** POST /technologies/:id/review */
+	async storeTechnologyReview({ params, request }) {
+		const data = request.only(['userId', 'content', 'rating', 'positive', 'negative']);
+
+		const review = {
+			content: data.content,
+			rating: data.rating,
+			positive: JSON.stringify(data.positive),
+			negative: JSON.stringify(data.negative),
+		};
+
+		const [technology, user] = await Promise.all([
+			Technology.findOrFail(params.id),
+			User.findOrFail(data.userId),
+		]);
+		const technologyReview = await TechnologyReview.create(review);
+		await Promise.all([
+			technologyReview.technology().associate(technology),
+			technologyReview.user().associate(user),
+		]);
```

Comment:
Como aqui há 3 operações no banco relacionadas a um mesmo fluxo, talvez seja interessante usar uma transação para garantir a atomicidade. Seria algo como (faz as adaptações aí...):

```suggestion
        let technologyReview;
		let trx;

		try {
			const { init, commit } = getTransaction();
			trx = await init();
		    
		    technologyReview = await TechnologyReview.create(review, trx);
    		await Promise.all([
			technologyReview.technology().associate(technology, trx), //não tenho ctz se a transação é o 2º param no associate...tem de confirmar.
			technologyReview.user().associate(user, trx),
		  ]);
		  
		  	await trx.commit();
		} catch (error) {
			await trx.rollback();
			throw error;
		}
		
		return technologyReview.toJSON();
```